### ISSUE TITLE
Feature proposal - Add participants info and lifecycle callbacks

### ISSUE THREAD

**Issue by marcuspuchalla** [fabianbormann/cardano-peer-connect#6]
It would be suitable for DApp creators and wallet providers to show the user more information about which p2p endpoint is connected. And also inform both endpoints about important lifecycle events.

I think for a great user experience when connecting wallets to DApps, these information and messages are missing:

1. After connecting to a DApp, the wallet has no information about the endpoint besides the cryptic endpoint address. No name or URL address that he has been connected to.

2. The connecting wallet has no information about the current status of the connection. If a wallet is closed, reopened, and connected to a DApp again, the DApp quietly logs this information that the client connection has already been established, but the client cannot know about that.

3. Once the connection has been established, the participants need to be informed if the other side is closing the connection. So the client shall inform the server when he wants to disconnect. And the server shall notify the client when he is shutting down. So both sides can use this information to inform the user about the connection's end.

**Comment by fabianbormann** [comment]
@marcuspuchalla I agree with you on all points and I really appreciate the lifecycle methods that you have already added 💯. With regards to the displayed information I thought about something like a blockie or a generated image so that a user can quickly verify that the image on wallet-side is the same as on dapp-side.

**Comment by marcuspuchalla** [comment]
> With regards to the displayed information I thought about something like a blockie or a generated image so that a user can quickly verify that the image on wallet-side is the same as on dapp-side.

I also thought about that 😄  Like in the TG Video Chat where there are 5 unique icons and everybody can instantly compare them. I could add something like this if you like at the end of the week probably.
If you are already on it, just let me know.

**Comment by fabianbormann** [comment]
Nice 👍 I have not started with the verification yet so it would be cool if you can look into this 💯  .. I would implement the dApp site (connect p2p wallet button and show up the QR code) for the CF react component: https://github.com/cardano-foundation/cardano-connect-with-wallet by end of this week
